cpp/src/structs/utilities.cu

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cudf/structs/structs_column_view.hpp>
+#include <cudf/utilities/error.hpp>
+
+namespace cudf {
+namespace structs {
+namespace detail {
+
+/**
+ * @copydoc cudf::structs::detail::extract_ordered_struct_children
+ *
+ */
+std::vector<std::vector<column_view>> extract_ordered_struct_children(
+  std::vector<column_view> const& struct_cols)
+{
+  // all inputs must be structs
+  CUDF_EXPECTS(
+    std::all_of(struct_cols.begin(),
+                struct_cols.end(),
+                [](column_view const& col) { return col.type().id() == type_id::STRUCT; }),
+    "Encountered non-struct input to extract_ordered_struct_children");
```

Comment:
I think this is redundant. When you construct a `structs_column_view` below it will perform the same check.

Reply:
Not quite.  The check at the top ensures all the inputs are structs.   The check internally checks that child(i) for each input are the same type.  So verifying the user didn't pass in some kind of weird set of mixed structs.

Reply:
When the `structs_column_view` is constructed here:
```
structs_column_view scv(struct_cols[col_index]);
```

It verifies that the input column's type is `STRUCT`: https://github.com/rapidsai/cudf/blob/5cf7106c1a86d49ae345eff31868343a894c864a/cpp/src/structs/structs_column_view.cu#L25

Reply:
Ah, right.